src/components/sections/Sponsor.section.tsx

```diff
@@ -1,199 +1,278 @@
+  Roomiez,
+  Balsamiq,
+  DigitalOcean,
+  Taskade,
+  Dcl,
+  StudentsUnion,
+  CCubed,
+  Mlh,
+  Fossa,
+  Lcs,
+} from "@assets";
 
 const SponsorFAQSection = () => {
-    return (
-        <div>
-            <section
-                id="sponsors-section"
-                className="sponsor-section relative z-20"
-            >
-                <img src={TopBorder} alt="Top Border" className="z-10 w-full" />
+  const carouselRefs = useRef([]);
+  useEffect(() => {
+    let animationFrameIds = new Map();
+
+    const initializeScrollAnimation = (carousel) => {
+      const totalAnimationTime = 8000; // Total cycle time for moving, excluding pauses
+      const pauseDuration = 2000; // Duration of pause at each end
+      let pauseScheduled = false;
+      let animationStartTime = Date.now() - pauseDuration; // Start with a pause
+      let lastAnimationPhase = "pause";
+
+      const handleScroll = () => {
+        const totalWidth = carousel.scrollWidth;
+        const visibleWidth = carousel.clientWidth;
+        const buffer = visibleWidth * 0.03; // Slightly less buffer for tighter right side
+        const scrollDistance = totalWidth - visibleWidth + buffer;
+
+        const easeInOutCubic = (t) =>
+          t < 0.5 ? 4 * t * t * t : 1 - Math.pow(-2 * t + 2, 3) / 2;
+
+        const updatePosition = () => {
+          const currentTime = Date.now();
+          const timeElapsedSinceStart = currentTime - animationStartTime;
+          const cycleDuration = totalAnimationTime + pauseDuration * 2; // Total duration including pauses
+          const cyclePosition = timeElapsedSinceStart % cycleDuration;
+
+          let currentPosition;
+
+          if (cyclePosition < pauseDuration) {
+            currentPosition = 0;
+            if (lastAnimationPhase !== "pause") {
+              pauseScheduled = false;
+            }
+          } else if (cyclePosition < totalAnimationTime / 2 + pauseDuration) {
+            const progress =
+              (cyclePosition - pauseDuration) / (totalAnimationTime / 2);
+            currentPosition = easeInOutCubic(progress) * scrollDistance;
+            lastAnimationPhase = "forward";
+          } else if (
+            cyclePosition <
+            totalAnimationTime / 2 + pauseDuration * 2
+          ) {
+            currentPosition = scrollDistance;
+            if (!pauseScheduled) {
+              lastAnimationPhase = "pause";
+              pauseScheduled = true;
+            }
+          } else {
+            const progress =
+              (cyclePosition - totalAnimationTime / 2 - pauseDuration * 2) /
+              (totalAnimationTime / 2);
+            currentPosition =
+              scrollDistance - easeInOutCubic(progress) * scrollDistance;
+            lastAnimationPhase = "backward";
+          }
+
+          carousel.style.transform = `translateX(-${currentPosition}px)`;
+          animationFrameIds.set(
+            carousel,
+            requestAnimationFrame(updatePosition)
```

Comment:
you know its real when you call `requestAnimationFrame`

Reply:
> you know its real when you call `requestAnimationFrame`

ong bro, 10x react dev right here